app/src/main/java/api/NetworkModule.kt

```diff
@@ -0,0 +1,60 @@
+package api
+
+import com.example.practice.BuildConfig
+import com.google.gson.GsonBuilder
+import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
+import kotlinx.coroutines.Deferred
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.http.GET
+
+
+/**
+ * UserApi is service interface which deals with user list data
+ */
+interface UserApi {
+    /**
+     * Function returns list of user list items asynchronously using 'Deferred'
+     */
+    @GET("users")
+    fun fetchUsersList() : Deferred<List<UsersListItem?>?>
```

Comment:
What is the difference between `Deferred` for this async request in comparison to `suspend` coroutine approach?

Reply:
@nets-dev
Q1] What is the difference between Deferred for this async request in comparison to suspend coroutine approach?
**Ans** :- In Deferred, as it is the async request we have to wait for the result using await(). But suspend approach make this thing super easy. I should use suspend instead of Deferred. Thank you for point this approach to me.

Reply:
You're welcome! There are a couple of comments on your answers below which you can take a look and reply. 
In the meantime, we'd like to setup a meeting with you tomorrow. Are you available 11 - 12:00 Finnish time? 
Please let us know below and we'll send you a meeting link. (**Note that the time is Finnish time**)

Meeting topics will be:

- Discussion on the project and comments in this review process 
- Brief overview of what our projects are and what sort of skills we find important 
- Discussion on what resources you've used while working on this project and any documentation, blog or website you found helpful 

How does that sound? @lnt-nishantdande

Reply:
@nets-dev Yes, I will be available between 11 -12:00 Finnish time i.e 1:30 PM - 2:30 PM IST.

Reply:
@lnt-nishantdande Great, here is the teams link for the meeting:
[Teams meeting 11 - 12:00](https://teams.microsoft.com/l/meetup-join/19%3ameeting_YmM1ZmRkY2UtOWZiNS00MjI0LWE2YzEtYjk4MGNmZmVhMDAw%40thread.v2/0?context=%7b%22Tid%22%3a%2279dc228f-c8f2-4016-8bf0-b990b6c72e98%22%2c%22Oid%22%3a%2213eab4a7-c748-4cb4-809e-27977c6812e2%22%7d)